lang/java/reef-io/src/main/java/org/apache/reef/io/network/util/Utils.java

```diff
@@ -31,30 +31,25 @@
 import java.util.Comparator;
 import java.util.List;
 
+/**
+ * Utility class for REEF IO network module.
+ */
 public final class Utils {
 
   private static final String DELIMITER = "-";
 
   /**
-   * TODO: Merge with parseListCmp() into one generic implementation.
-   */
-  public static List<Identifier> parseList(
-      final String ids, final IdentifierFactory factory) {
-    final List<Identifier> result = new ArrayList<>();
-    for (final String token : ids.split(DELIMITER)) {
-      result.add(factory.getNewInstance(token.trim()));
-    }
-    return result;
-  }
-
-  /**
-   * TODO: Merge with parseList() into one generic implementation.
+   * Parse a string of multiple IDs.
+   *
+   * @param ids A string containing multiple IDs
+   * @param factory An Identifier factory
+   * @param <T> A type
+   * @return A list of identifier
    */
-  public static List<ComparableIdentifier> parseListCmp(
```

Comment:
We can't delete public method in public class without going through deprecation. If we believe all methods in the class are unused, we should deprecate the whole class. Otherwise we need to mark parseListCmp as deprecated and to delete it in next release.